pkg/resolution/resolver/git/config.go

```diff
@@ -43,3 +51,54 @@ const (
 	// APISecretNamespaceKey is the config map key for the token secret's namespace
 	APISecretNamespaceKey = "api-token-secret-namespace"
 )
+
+type GitResolverConfig struct {
+	ScmTokens map[string]ScmInfo
+}
+
+type ScmInfo struct {
+	Timeout            string `json:"fetch-timeout"`
+	URL                string `json:"default-url"`
+	Revision           string `json:"default-revision"`
+	Org                string `json:"default-org"`
+	ServerURL          string `json:"server-url"`
+	SCMType            string `json:"scm-type"`
+	APISecretName      string `json:"api-token-secret-name"`
+	APISecretKey       string `json:"api-token-secret-key"`
+	APISecretNamespace string `json:"api-token-secret-namespace"`
+}
+
+func GetGitConfig(ctx context.Context) GitResolverConfig {
```

Comment:
This function needs unit tests!

Reply:
i tried to do this, but was not able to mock, may be will give one more try.

Reply:
unit tests added